src/ProductConstructionService/ProductConstructionService.SubscriptionTriggerer/SubscriptionTriggerer.cs

```diff
@@ -14,36 +15,49 @@ public class SubscriptionTriggerer
 {
     private readonly ILogger<SubscriptionTriggerer> _logger;
     private readonly BuildAssetRegistryContext _context;
-    private readonly IWorkItemProducerFactory _workItemProducerFactory;
+    private readonly IWorkItemProducerFactory _defaultWorkItemProducerFactory;
+    private readonly IWorkItemProducerFactory _codeflowWorkItemProducerFactory;
     private readonly SubscriptionIdGenerator _subscriptionIdGenerator;
 
     public SubscriptionTriggerer(
         ILogger<SubscriptionTriggerer> logger,
         BuildAssetRegistryContext context,
-        IWorkItemProducerFactory workItemProducerFactory,
+        [FromKeyedServices(WorkItemConfiguration.DefaultWorkItemType)] IWorkItemProducerFactory defaultWorkItemProducerFactory,
+        [FromKeyedServices(WorkItemConfiguration.CodeflowWorkItemType)] IWorkItemProducerFactory codeflowWorkItemProducerFactory,
         SubscriptionIdGenerator subscriptionIdGenerator)
     {
         _logger = logger;
         _context = context;
-        _workItemProducerFactory = workItemProducerFactory;
+        _defaultWorkItemProducerFactory = defaultWorkItemProducerFactory;
+        _codeflowWorkItemProducerFactory = codeflowWorkItemProducerFactory;
         _subscriptionIdGenerator = subscriptionIdGenerator;
     }
 
     public async Task TriggerSubscriptionsAsync(UpdateFrequency targetUpdateFrequency)
     {
-        var workItemProducer = _workItemProducerFactory.CreateProducer<SubscriptionTriggerWorkItem>();
+        var defaultWorkItemProducer = _defaultWorkItemProducerFactory.CreateProducer<SubscriptionTriggerWorkItem>();
+        var codeflowWorkItemProducer = _codeflowWorkItemProducerFactory.CreateProducer<SubscriptionTriggerWorkItem>();
```

Comment:
I wonder how much easier would it be to just return the right producer from the factory. Technically the `if/else` would be implemented just once there while we can have any number of places producing these work items all around the codebase which now would have to have the `if/else` every time

Reply:
The factory is already configured to a queue, so it wouldn't work in the current implementation, we'd need a `FactoryFactory` 😅

Reply:
oh or no, we just pass the queueName to the `IWorkItemProducer`

Reply:
Yeah, I don't see why it wouldn't know about both and accept a parameter `isCodeFlowSubscription` or something. We wouldn't have to inject two factories everywhere then